addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/handler/ModbusDataThingHandler.java

```diff
@@ -861,6 +873,21 @@ private void tryUpdateState(ChannelUID uid, State state) {
         }
     }
 
+    private void whenExpired(long now, ChannelUID uid, State state, BiConsumer<ChannelUID, State> action) {
+        @Nullable
+        State lastState = channelLastState.get(uid);
+        long lastUpdatedMillis = channelLastUpdated.getOrDefault(uid, 0L);
+        long millisSinceLastUpdate = now - lastUpdatedMillis;
+        if (lastUpdatedMillis <= 0L || lastState == null || updateUnchangedValuesEveryMillis <= 0L
+                || millisSinceLastUpdate > updateUnchangedValuesEveryMillis || !lastState.equals(state)) {
+            try {
+                action.accept(uid, state);
+            } finally {
+                channelLastUpdated.put(uid, now);
```

Comment:
Mainly curious why from the finally if the update fails to persist (ie action.accept throw an exception) it would imho make sense not to update the last updated.

Reply:
@martinvw : good observation... I agree -- would make sense not to update last updated, and try again next time.

In practice, exceptions are probably not thrown.

Will fix this